src/cpp/src/text2image/models/sd3_transformer_2d_model.cpp

```diff
@@ -0,0 +1,134 @@
+    read_json_param(data, "attention_head_dim", attention_head_dim);
+    read_json_param(data, "num_attention_heads", num_attention_heads);
+    read_json_param(data, "joint_attention_dim", joint_attention_dim);
+    read_json_param(data, "caption_projection_dim", caption_projection_dim);
+    read_json_param(data, "pooled_projection_dim", pooled_projection_dim);
+    read_json_param(data, "out_channels", out_channels);
+    read_json_param(data, "pos_embed_max_size", pos_embed_max_size);
+
+    file.close();
+
+    // block_out_channels should be read from VAE encoder / decoder config to compute proper m_vae_scale_factor
+    std::filesystem::path vae_config_path = config_path.parent_path().parent_path() / "vae_decoder" / "config.json";
+    file.open(vae_config_path);
+    OPENVINO_ASSERT(file.is_open(), "Failed to open ", vae_config_path);
+    data = nlohmann::json::parse(file);
+    read_json_param(data, "block_out_channels", block_out_channels);
+}
+
+SD3Transformer2DModel::SD3Transformer2DModel(const std::filesystem::path& root_dir) :
+    m_config(root_dir / "config.json") {
+    m_model = utils::singleton_core().read_model((root_dir / "openvino_model.xml").string());
+
+    // compute VAE scale factor
+    m_vae_scale_factor = std::pow(2, m_config.block_out_channels.size() - 1);
+    m_patch_size = m_config.patch_size;
+}
+
+SD3Transformer2DModel::SD3Transformer2DModel(const std::filesystem::path& root_dir,
+                const std::string& device,
+                const ov::AnyMap& properties) :
+    SD3Transformer2DModel(root_dir) {
+    compile(device, properties);
+}
+
+SD3Transformer2DModel::SD3Transformer2DModel(const SD3Transformer2DModel&) = default;
+
+const SD3Transformer2DModel::Config& SD3Transformer2DModel::get_config() const {
+    return m_config;
+}
+
+// TODO:
+SD3Transformer2DModel& SD3Transformer2DModel::reshape(int batch_size, int height, int width, int tokenizer_model_max_length) {
+    OPENVINO_ASSERT(m_model, "Model has been already compiled. Cannot reshape already compiled model");
+
+    //hidden_states=latent_model_input,
+    //timestep=timestep,
+    //encoder_hidden_states=prompt_embeds,
+    //pooled_projections=pooled_prompt_embeds,
+
+    height /= m_vae_scale_factor;
+    width /= m_vae_scale_factor;
+
+    std::map<std::string, ov::PartialShape> name_to_shape;
+
+    for (auto && input : m_model->inputs()) {
+        std::string input_name = input.get_any_name();
+        name_to_shape[input_name] = input.get_partial_shape();
+        if (input_name == "timestep") {
+            name_to_shape[input_name][0] = batch_size;
+        } else if (input_name == "hidden_states") {
+            name_to_shape[input_name] = {batch_size, name_to_shape[input_name][1], height, width};
+        } else if (input_name == "encoder_hidden_states") {
+            name_to_shape[input_name][0] = batch_size;
+            name_to_shape[input_name][1] = tokenizer_model_max_length * 2;
```

Comment:
where `2` is taken from ? can we compute this constant somehow ? it does not look like batch size multiplier ..

Reply:
We're getting this `x2` because of the concatenation of `prompt_embed` and `t5_prompt_embed` (same for negative embeds): https://github.com/huggingface/diffusers/blob/743a5697f2596567c991e8bc5dd2d4d4a4fffa99/src/diffusers/pipelines/stable_diffusion_3/pipeline_stable_diffusion_3.py#L440

Reply:
ok, let's later add this comment to the code as it seems useful